### ISSUE TITLE
Question about the runtime of main experiments in the RLPR paper

### ISSUE THREAD

**Issue by hcx-stu** [OpenBMB/RLPR#16]
Thank you for open-sourcing the code for RLPR. I’ve been studying the implementation with great interest, and I really appreciate the clarity of your work.

According to the paper, the experiments were conducted using 32 NVIDIA A100 GPUs. I was wondering if you could kindly share the approximate training time (or total runtime) for the main experiments reported in the paper. This information would be extremely helpful for reproducing the results.

**Comment by hcx-stu** [comment]
Furthermore, I would like to ask at which training step the results reported in the main experiments were obtained.

**Comment by jibo27** [comment]
Thank you for your interest! Our final Qwen model underwent 4 days and 13 hours of training (from step 0 to step 320). The figure below shows the runtime (in seconds) across training steps. Periodic peaks occur due to evaluation every 40 steps. We selected the 320-step checkpoint as it achieved optimal performance.

<img width="1444" height="824" alt="Image" src="https://github.com/user-attachments/assets/13650a1c-55ad-49fa-a747-404b73960084" />

**Comment by hcx-stu** [comment]
> Thank you for your interest! Our final Qwen model underwent 4 days and 13 hours of training (from step 0 to step 320). The figure below shows the runtime (in seconds) across training steps. Periodic peaks occur due to evaluation every 40 steps. We selected the 320-step checkpoint as it achieved optimal performance.
> 
> <img alt="Image" width="1444" height="824" src="https://private-user-images.githubusercontent.com/40049332/475934110-13650a1c-55ad-49fa-a747-404b73960084.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ2NDUxMDAsIm5iZiI6MTc1NDY0NDgwMCwicGF0aCI6Ii80MDA0OTMzMi80NzU5MzQxMTAtMTM2NTBhMWMtNTVhZC00OWZhLWE3NDctNDA0YjczOTYwMDg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA4VDA5MjAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlMGRmYzViNzVjODdhMjdiMzJjNTUzZmRiMmQ5ODM2MWY2NTE3YWY5YzkwZGYxNzVmZTYxZGYyODczZWQ1YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q9Kgd77Abi5O6au2Xl-cexNmqv49aa6nDobosCwhpz4">

Thank you very much for your reply. We reproduced the experiment using 48 NVIDIA H100 GPUs, but unfortunately, our results did not meet expectations. We ran the experiment for 320 steps and observed the best performance at step 200; however, there is still a substantial gap compared to the results reported in the paper. Could you kindly share any insights into what might be causing this discrepancy?

| Step | MMLU-Pro | GPQA  | TheoremQA | WebInst. | MATH-500 | Minerva | AIME 24 | General | All   |
|------|----------|-------|-----------|----------|----------|---------|---------|---------|-------|
| 200  | 53.15    | 35.98 | 56.09     | 74.84    | 76.70    | 54.23   | 15.21   | 55.02   | 52.32 |
| 320  | 52.75    | 34.22 | 53.48     | 73.82    | 74.80    | 50.83   | 13.75   | 53.57   | 50.52 |

**Comment by jibo27** [comment]
Hi, you can first check whether the evaluation was conducted using **LLM-as-a-judge** or a **rule-based approach**. If the evaluation was not performed using LLM-as-a-judge, the scores may appear lower than expected, as some correct responses might be mistakenly marked as incorrect.  

You can review the model's actual outputs and their correctness judgments in the saved test logs: `data/logs/test_generations_*`.  

For reference, here is the [WandB report link](https://api.wandb.ai/links/THUNLP-MM/jbbolsar) containing the full metrics from our model training. In this report, the performance was measured using **LLM-as-a-judge** (specifically, a Qwen instruct model).  The final results were further refined based on manual verification of the outputs.

**Comment by hcx-stu** [comment]
> Hi, you can first check whether the evaluation was conducted using **LLM-as-a-judge** or a **rule-based approach**. If the evaluation was not performed using LLM-as-a-judge, the scores may appear lower than expected, as some correct responses might be mistakenly marked as incorrect.
> 
> You can review the model's actual outputs and their correctness judgments in the saved test logs: `data/logs/test_generations_*`.
> 
> For reference, here is the [WandB report link](https://api.wandb.ai/links/THUNLP-MM/jbbolsar) containing the full metrics from our model training. In this report, the performance was measured using **LLM-as-a-judge** (specifically, a Qwen instruct model). The final results were further refined based on manual verification of the outputs.

Thank you for your reply. We employed LLM-as-a-judge (Qwen/Qwen2.5-72B-Instruct) for evaluation. Although our experimental results differ somewhat from those reported in the paper, they nonetheless further substantiate the effectiveness of RLPR. I look forward to seeing your work published in the near future.

**Comment by PangziZhang523** [comment]
@jibo27 Hello, have you experimented with longer caption-like general domain tasks?

**Comment by jibo27** [comment]
> [@jibo27](https://github.com/jibo27) Hello, have you experimented with longer caption-like general domain tasks?

Hi, we didn't test much on tasks like these, but this is an interesting direction!

**Comment by hcx-stu** [comment]
[@jibo27](https://github.com/jibo27) Hello, I have a question regarding the RLVR baseline mentioned in the paper. Could you please clarify what training data was used for RLVR?

Thank you in advance for your time and support! 🙏

**Comment by jibo27** [comment]
> [@jibo27](https://github.com/jibo27) Hello, I have a question regarding the RLVR baseline mentioned in the paper. Could you please clarify what training data was used for RLVR?
> 
> Thank you in advance for your time and support! 🙏

Hi, the training data is the same as our training data, i.e., [RLPR-Train-Dataset](https://huggingface.co/datasets/openbmb/RLPR-Train-Dataset).

**Comment by hcx-stu** [comment]
[@jibo27](https://github.com/jibo27) Hello, sorry to bother you.

I noticed that the RLPR paper mentions reproducing the VeriFree work. We also attempted to replicate VeriFree, but unfortunately, we encountered an issue where the model collapses after around 1000 steps and starts producing nonsensical outputs. We reached out to the authors of VeriFree, but unfortunately, we did not receive a response.

Since I saw that you have also worked on reproducing related work, I wanted to ask if you’ve encountered a similar issue? If possible, would you be willing to share the VeriFree model you trained? It would be greatly appreciated!

Thank you so much!

**Comment by zyandtom** [comment]
> > Thank you for your interest! Our final Qwen model underwent 4 days and 13 hours of training (from step 0 to step 320). The figure below shows the runtime (in seconds) across training steps. Periodic peaks occur due to evaluation every 40 steps. We selected the 320-step checkpoint as it achieved optimal performance.
> > <img alt="Image" width="1444" height="824" src="https://private-user-images.githubusercontent.com/40049332/475934110-13650a1c-55ad-49fa-a747-404b73960084.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ2NDUxMDAsIm5iZiI6MTc1NDY0NDgwMCwicGF0aCI6Ii80MDA0OTMzMi80NzU5MzQxMTAtMTM2NTBhMWMtNTVhZC00OWZhLWE3NDctNDA0YjczOTYwMDg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA4VDA5MjAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlMGRmYzViNzVjODdhMjdiMzJjNTUzZmRiMmQ5ODM2MWY2NTE3YWY5YzkwZGYxNzVmZTYxZGYyODczZWQ1YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q9Kgd77Abi5O6au2Xl-cexNmqv49aa6nDobosCwhpz4">
> 
> Thank you very much for your reply. We reproduced the experiment using 48 NVIDIA H100 GPUs, but unfortunately, our results did not meet expectations. We ran the experiment for 320 steps and observed the best performance at step 200; however, there is still a substantial gap compared to the results reported in the paper. Could you kindly share any insights into what might be causing this discrepancy?
> 
> Step	MMLU-Pro	GPQA	TheoremQA	WebInst.	MATH-500	Minerva	AIME 24	General	All
> 200	53.15	35.98	56.09	74.84	76.70	54.23	15.21	55.02	52.32
> 320	52.75	34.22	53.48	73.82	74.80	50.83	13.75	53.57	50.52

so although it not reach the metric in paper, is it better than RLVR or model-based reward?

**Comment by DanielSHKao** [comment]
> Thank you for your interest! Our final Qwen model underwent 4 days and 13 hours of training (from step 0 to step 320). The figure below shows the runtime (in seconds) across training steps. Periodic peaks occur due to evaluation every 40 steps. We selected the 320-step checkpoint as it achieved optimal performance.
> 
> <img alt="Image" width="1444" height="824" src="https://private-user-images.githubusercontent.com/40049332/475934110-13650a1c-55ad-49fa-a747-404b73960084.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODEwMDUxOTEsIm5iZiI6MTc4MTAwNDg5MSwicGF0aCI6Ii80MDA0OTMzMi80NzU5MzQxMTAtMTM2NTBhMWMtNTVhZC00OWZhLWE3NDctNDA0YjczOTYwMDg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNjA5VDExMzQ1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1MThiMThmODlhYWY0MThhZTRkYjRhOTM3M2U0MzA5NzNiMDIxNmQwNjI2NzI4YjIyOWU3ZTZjZjdiMzQxODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.PYGS9vcP-AnOIVBfb9I9uF2gQyFZn0LI3g4CEvHMhkU">

Hi Author, 

I found your work really interesting. I would like to cite your paper and extend it for my future project. However, I also failed to reproduce the numbers on the paper. My replication results are very close to @hcx-stu 's. 

I somehow consider the way you picked the best number with one run from the test sets was kind of cherry picking, as the training recipe is supposed to be blind to the testing benchmarks... Would you mind sharing the average performance for multiple runs (3 for example)? Thanks!
